File: src/main/java/com/youdeyiwu/enums/forum/CommentReviewStateEnum.java
package com.youdeyiwu.enums.forum;

import lombok.Getter;

/**
 * comment review state.
 *
 * @author dafengzhen
 */
@Getter
public enum CommentReviewStateEnum {

  /**
   * approved (Default).
   */
  APPROVED,

  /**
   * rejected (The forum administrators, section administrators, and comment authors have access).
   */
  REJECTED,

  /**
   * pendingReview (The forum administrators,
   * section administrators, and comment authors have access).
   */
  PENDING_REVIEW,

}

File: app/lib/enums/comment_review_state_enum.dart
import 'package:json_annotation/json_annotation.dart';

enum CommentReviewStateEnum {
  /// approved (Default).
  @JsonValue('APPROVED')
  approved,

  /// rejected (The forum administrators, section administrators, and comment authors have access).
  @JsonValue('REJECTED')
  rejected,

  /// pending review (The forum administrators,
  /// section administrators, and comment authors have access).
  @JsonValue('PENDING_REVIEW')
  pendingReview,
}
